pkg/account/main.go

```diff
@@ -46,6 +47,21 @@ func getMain(jaeger *v1.Jaeger) *corev1.ServiceAccount {
 }
 
 // JaegerServiceAccountFor prints service name for Jaeger instance
-func JaegerServiceAccountFor(jaeger *v1.Jaeger) string {
-	return fmt.Sprintf("%s", jaeger.Name)
+func JaegerServiceAccountFor(jaeger *v1.Jaeger, component string) string {
+	sa := ""
+	switch component {
+	case "collector":
```

Comment:
It would probably make sense to use a typed constant here, like, `CollectorComponent Component` and change the func's signature to `component Component`

Reply:
added new type/const